servicetalk-grpc-protoc/src/main/java/io/servicetalk/grpc/protoc/Generator.java

```diff
@@ -214,12 +220,17 @@ private void addSerializationProviderInit(final State state, final TypeSpec.Buil
                         builder, t, t));
 
         staticInitBlockBuilder
-                .addStatement("$L = $L.build()", serializationProvider, builder)
+                .addStatement("return $L.build()", builder)
                 .build();
 
         serviceClassBuilder
-                .addField(GrpcSerializationProvider, serializationProvider, PRIVATE, STATIC, FINAL)
-                .addStaticBlock(staticInitBlockBuilder.build());
+                .addMethod(methodBuilder(initSerializationProvider)
+                        .addModifiers(STATIC)
+                        .returns(GrpcSerializationProvider)
+                        .addParameter(GrpcSupportedEncodings, supportedEncodings, FINAL)
```

Comment:
Since now we are creating this serialization provider individually for each route, we can restrict the registrations to only the types required by that route. wdyt?

If you agree, it is OK to do that as a follow up

Reply:
I agree, I was also thinking of caching the providers when supported-encodings are the same, I will open an issue to add this work separately of this PR.